helix-term/src/ui/mod.rs

```diff
@@ -276,6 +279,74 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> FilePi
     picker
 }
 
+type FileBrowser = Picker<(PathBuf, bool), (PathBuf, Style)>;
+
+pub fn file_browser(root: PathBuf, editor: &Editor) -> Result<FileBrowser, std::io::Error> {
+    let directory_style = editor.theme.get("ui.text.directory");
+    let directory_content = directory_content(&root)?;
+
+    let columns = [PickerColumn::new(
+        "path",
+        |(path, is_dir): &(PathBuf, bool), (root, directory_style): &(PathBuf, Style)| {
+            let name = path.strip_prefix(root).unwrap_or(path).to_string_lossy();
+            if *is_dir {
+                Span::styled(format!("{}/", name), *directory_style).into()
+            } else {
+                name.into()
+            }
+        },
+    )];
+    let picker = Picker::new(
+        columns,
+        0,
+        directory_content,
+        (root, directory_style),
+        move |cx, (path, is_dir): &(PathBuf, bool), action| {
+            if *is_dir {
+                let owned_path = path.clone();
```

Comment:
```suggestion
                let new_root = helix_stdx::path::normalize(path);
```

when switching paths we should normalize the input path because it may be `path.join("..")` and `owned_path.parent()` returns `path` (https://play.rust-lang.org/?version=stable&mode=debug&edition=2021). The effect is that if you use the file browser to try to reach the file system root then you'll still see `../` in the options unless we normalize the new root here

Reply:
Yes, that was the reason why I canonicalized the path before =) I was not aware of the normalize utility function, thanks!